test/core/src/lombok/core/TestHandlerUtil.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2015 The Project Lombok Authors.
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package lombok.core;
+
+import static lombok.core.handlers.HandlerUtil.buildAccessorName;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import org.junit.Test;
+
+public class TestHandlerUtil {
+	
+	@Test
+	public void testBuildAccessorName() {
+		
+		/* normal cases */
+		assertEquals("setFieldName", buildAccessorName("set", "fieldName"));
+		assertEquals("setI", buildAccessorName("set", "i"));
+		assertEquals("setI", buildAccessorName("set", "I"));
+		assertEquals("setUrl", buildAccessorName("set", "Url"));
+		
+		/* when the second character is uppercase, leave the first character unchanged */
```

Comment:
This can also be shifted to a new method
`testBuildAccessorNameWithUpperCaseSecondCharacterInSuffix()`